petergreve/week2/Ass3/index.html

```diff
@@ -0,0 +1,233 @@
+<!DOCTYPE html>
+<html>
+<head>
+	<title>Deep dive into React</title>
+	<script src="http://fb.me/react-0.14.7.js"></script>
+	<script src="http://fb.me/react-dom-0.14.7.js"></script>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.23/browser.min.js"></script>
+</head>
+<body>
+
+<div id="root"> </div>
+	<script type="text/babel">
+
+	var List = React.createClass({
+
+		handleInput: function() {
+		  	var message = this.input.value;
+		  	this.props.createMessage(message,this.props.list.listId);
+		},
+		render: function() {
+			return 	<div>
+						{this.props.list.listId} - {this.props.list.listName}
+						<br />
+						<input type="text" ref={(component) => this.input = component} /><button type="button" onClick={this.handleInput}>Submit Message</button>
+						<Messages
+						 messages={this.props.messages} 
+						 archived={this.props.archived} 
+						 archiveMessage={this.props.archiveMessage}
+						 deleteMessage={this.props.deleteMessage} 
+						 moveMessage={this.props.moveMessage} />
+					</div>
+		}
+
+	})
+
+	var Messages = React.createClass({
+
+		handleDeleteClick: function(message) {
+			this.props.deleteMessage(message.messageId);
+		},
+		handleArchiveClick: function(message) {
+			this.props.archiveMessage(message.messageId, true);
+		},
+		handleUnarchiveClick: function(message) {
+			this.props.archiveMessage(message.messageId, false);
+		},
+		handleDownClick: function(message) {
+			this.props.moveMessage(message.messageId, true);
+		},
+		handleUpClick: function(message) {
+			this.props.moveMessage(message.messageId, false);
+		},
+		render: function () {
+
+			var style = {
+	   			color: 'blue',
+    		};
+
+			var that = this;
+			return 	<div>
+					    <ol>
+					   		{this.props.messages.map(function(message,i) {
+					     		return <div key={i}>
+					     					{message.text}
+					     					<button type="button" onClick={that.handleDeleteClick.bind(null, message)}>Delete</button>
+					     					<button type="button" onClick={that.handleArchiveClick.bind(null, message)}>Archive</button>
+					     					<button type="button" onClick={that.handleDownClick.bind(null, message)}>Move Down</button>
+					     					<button type="button" onClick={that.handleUpClick.bind(null, message)}>Move Up</button>
+					     				</div>;
+					     	})}
+					    </ol>
+					    <ol>
+					   		{this.props.archived.map(function(message,i) {
+					     		return <div style={style} key={i}>
+					     					{message.text}
+					     					<button type="button" onClick={that.handleUnarchiveClick.bind(null, message)}>Unarchive</button>
+					     				</div>;
+					     	})}
+					    </ol>
+      				</div>
+		}
+
+	});
+
+	var ErrorMessage = React.createClass({
+		render: function () {
+
+			var style = {
+	   			color: 'red',
+    		};
+			return 	<div style={style}>
+						{this.props.errorMessage}
+					</div>
+		}
+
+	});
+
+	var App = React.createClass({
+		getInitialState: function() {
+		    return {
```

Comment:
Good idea to show me your data-model!

But of cause nothing you would have in production, I hope :)